From 7ef57e2b225878d8b62a701944cd8c9a37f37991 Mon Sep 17 00:00:00 2001
From: arunthomas <arunthomasb@ami.com>
Date: Fri, 19 May 2023 18:43:13 +0530
Subject: [PATCH 29/29] Fix for SSE URI returning 405 Error

/redfish/v1/EventService/SSE URI used to return 405 Method Not Allowed error when accessed.
This is because invalid operands were used to check the allowable methods.

Tested: Able to launch the /redfish/v1/EventService/SSE URI from browser and recieved event notifications.

Signed-off-by: arunthomas <arunthomasb@ami.com>
---
 http/routing.hpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/http/routing.hpp b/http/routing.hpp
index 4299d6fb..b92f37bb 100644
--- a/http/routing.hpp
+++ b/http/routing.hpp
@@ -1475,8 +1475,9 @@ class Router
             throw std::runtime_error("Trie internal structure corrupted!");
         }
 
-        if ((rules[ruleIndex]->getMethods() &
-             (1U << static_cast<size_t>(req.method()))) == 0)
+        BaseRule& rule = *rules[ruleIndex];
+        size_t methods = rule.getMethods();
+        if ((methods & (1U << static_cast<size_t>(*verb))) == 0)
         {
             BMCWEB_LOG_DEBUG << "Rule found but method mismatch: " << req.url
                              << " with " << req.methodString() << "("
-- 
2.25.1

